doc/release/1.11.0-notes.rst

```diff
@@ -118,3 +118,12 @@ c_contiguous arrays at some future time. A work around that is backward
 compatible is to use `a.T.view(...).T` instead. A parameter will also be
 added to the view method to explicitly ask for Fortran order views, but
 that will not be backward compatible.
+
+Invalid arguments to ```array.flatten```
```

Comment:
It is more general, this is used (almost) everywhere where an `order` argument is used. Also, I think you don't have to argue that it should not be allowed, it will not be allowed in the future ;). Anyway, the document will be revised again for the release.

When you are done with this, could you squash the commits together into one or two (`rebase -i HEAD~7` or similar for example). Best give it a commit like "DEP: ..." and you can include "closes gh-6598" at the end in the message as well (the message will be closed automatically and we have the reference).